pulsar/error.go

```diff
@@ -0,0 +1,104 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package pulsar
+
+import "C"
```

Comment:
`import C`? we should remove it?

I am thinking about another question, do we need to define Result or error here, because this looks more like the style of the c language, but not the style of the go language. Of course, the error in the go language itself does not support stack return. Do we consider using [pkg-errors](https://github.com/pkg/errors) or [juju-errors] (https://github.com/juju/errors)

Reply:
Ups, "import C" was a leftover (the API file were imported from the current library). 

Regarding exposing Result, the idea is more to give an option to downcast `error` to a `pulsar.Error` and retrieve the exact code, maybe to give the chance to application to handle different errors in different ways. It's not strictly necessary per-se. The important part is, at least initially, that we can print a meaningful error message.

Reply:
ok, I agree with your point of view, then we are considering encapsulating the information of these error codes into the third-party Error library, which guarantees that we have only one error in the global use.